Feature/blog highlight section categories

mahdisolgi1 opened the pull request:
This PR introduces two major updates to the BlogHighlightSection component:

1. Fetch Categories Dynamically :

- Replaces the hardcoded categories with dynamic fetching from the server. The categories are now fetched using the useGetCategories hook and displayed accordingly in the component.

2. Error Handling for Fetching Categories:

- Implements error handling for the category fetching process. If the categories fail to load due to an error, an appropriate message will be logged, ensuring the component remains functional even in the case of API failures.

mahdisolgi1 committed:
refactor: Move Resources component into Header for logical grouping

- Moved `Resources` component from `heroSection` on the homepage to the `Header` component.
- Removed `Resources` import from the homepage as it now belongs within the `Header` structure.
- This change improves component organization by grouping `Resources` within `Header`, as they are conceptually related.
- Updated relevant styles and layout adjustments to ensure smooth integration of `Resources` within `Header`.

mahdisolgi1 committed:
style: Make Header component responsive

- Adjusted layout and styling of the Header component for various screen sizes.

mahdisolgi1 committed:
feat(Resources): make responsive for laptop screens

mahdisolgi1 committed:
feat(PerformanceOverview): make responsive for laptop screens

mahdisolgi1 committed:
feat(Resources): make FutureTechnologyCards responsive for laptop screens and adjust layout

mahdisolgi1 committed:
feat(BlogHighlightSection): make BlogHighlightSection responsive for laptop screens and adjust layout

mahdisolgi1 committed:
feat(EbookHero): make EbookHero responsive for laptop screens and fix a small bad styling

mahdisolgi1 committed:
feat(EbookHero): replace EbookHero images with the ones with better quality and also fix a small bad styling for them

mahdisolgi1 committed:
feat(Testimonials): make Testimonials responsive for laptop screens

mahdisolgi1 committed:
feat(FutureTechRevolutionSection): make FutureTechRevolutionSection responsive for laptop screens and adjust layout

mahdisolgi1 committed:
feat(Footer): add responsive styles for Footer on laptop and adjust layout

mahdisolgi1 committed:
fix(FutureTechnologyCards): correct minor styling issues

mahdisolgi1 committed:
feat(BlogHighlightSection): synchronize articles with selected category

- Added conditional rendering to ensure articles are updated based on the selected category
- If the category changes, fetch and display articles from the same category

mahdisolgi1 committed:
refactor(BlogHighlightSection): reuse the active state styles without needing to add the active class to each individual child

mahdisolgi1 committed:
feat(TopBanner): add desktop responsiveness

mahdisolgi1 committed:
feat(Nav): add desktop responsiveness

mahdisolgi1 committed:
feat(Header): add responsive desktop styles for Header component

mahdisolgi1 committed:
feat(Resources): add responsive desktop styles for Resources component used in Header component

mahdisolgi1 committed:
feat(PerformanceOverview): add responsive desktop styles for PerformanceOverview component

mahdisolgi1 committed:
feat(FutureTechnologyCards): add responsive desktop styles for FutureTechnologyCards component

mahdisolgi1 committed:
feat(BlogHighlightSection): add responsive desktop styles for BlogHighlightSection component

mahdisolgi1 committed:
feat(BlogHighlightSection): fetch categories from server and display dynamically instead of hardcoded values

mahdisolgi1 committed:
feat(BlogHighlightSection): add error handling for fetching categories from server

amirrezaDev1378 commented on frontend/src/components/BlogHighlightSection.tsx:
simplify this

  if (!res) {
         return;
      }
      if (articleIsLoading) {
         return;
      }
      if (articleError) {
         console.error(articleError);
         return;
      }

mahdisolgi1 replied:
I simplified the logic  in a7cee89 commit check that out

mahdisolgi1 committed:
chore(BlogHighlightSection): simplify logic for article filtering and display

- Improved readability and maintainability in articles filtering
- Streamlined the topic click handler by removing redundant code
